FreeRTOS/FreeRTOS-Kernel
# Make the type used to hold run-time counter values configurable

Description
-----------
Introduce configRUN_TIME_COUNTER_TYPE which enables developers to define the type used to hold run time statistic counters.  Defaults to uint32_t for backward compatibility.  #define configRUN_TIME_COUNTER_TYPE to a type (for example, uint64_t) in FreeRTOSConfig.h to override the default.

Introduce ulTaskGetIdleRunTimePercent() to complement the pre-existing ulTaskGetIdleRunTimeCounter().  Whereas the pre-existing function returns the raw run time counter value, the new function returns the percentage of the entire run time consumed by the idle task.  Note the amount of idle time is only a good measure of the slack time in a system if there are no other tasks executing at the idle priority, tickless idle is not used, and configIDLE_SHOULD_YIELD is set to 0.

Test Steps
-----------
Somewhat tested by the Win32 demo in the FreeRTOS/FreeRTOS repo which prints the amount of time used by the idle task in both its comprehensive and blinky demos.

Related Issue
-----------
Corresponding updates to the Win32Demo in the FreeRTOS/FreeRTOS repo are in a separate PR that will be created after this one.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
